Contact form section

Nathan-h05 opened the pull request:
## 🔧 Description
- Add public Contact form (split layout: form left, image right) with required fields: Full Name, Email, Request Type, Additional details
- Implement backend POST /api/contact with validation, email send via reusable Nodemailer helper, and reply-to submitter
- Add IP-based rate limiter (10 req/min) on contact endpoint
- Integrate optional Cloudflare Turnstile captcha (frontend flag off for now; backend toggled via CONTACT_REQUIRE_CAPTCHA)

## ✅ Related Issues
N/A (feature addition)

## 📝 Improvements
- Refactor mailer into module-level transporter + sendEmail helper
- Add simple HTML email template helper for consistent formatting
- Lazy-load captcha only on interaction/viewport to reduce overhead
- Clearer error handling and user feedback (loading/success/error states)

## 🧪 Testing
- [x] Code runs locally
- [x] Rate limit triggers on rapid repeated submits (429)
- [x] With CONTACT_REQUIRE_CAPTCHA=true and frontend flag on, captcha is required
- [x] With CONTACT_REQUIRE_CAPTCHA=false and frontend flag off, submit works without captcha

## 📸 Screenshots (if UI changes)
<img width="1354" height="845" alt="image" src="https://github.com/user-attachments/assets/bf78cf46-40a9-4e86-8094-6717822e7ac5" />

## 🚀 Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

Nathan-h05 committed:
public contact form + email sending

- Frontend: new ContactForm (required fields, split layout), wired to backend with loading/success/error
- Backend: POST /api/contact with validation, reusable Nodemailer mailer, simple HTML template, reply-to submitter
- Note: rate limiting and captcha to be added later

Nathan-h05 committed:
Add Turnstile captcha and rate limiting to contact form

- Backend: IP-based rate limiter (10 req/min) on POST /api/contact
- Backend: Cloudflare Turnstile verification
- Frontend: Turnstile widget integrated; token sent with form
- UX: Lazy-load captcha (on interaction/viewport) and enable submit on solved captcha

Nathan-h05 committed:
Added a USE_CAPTCHA flag to disable/enable captcha

Nathan-h05 committed:
Added CONTACT_REQUIRE_CAPTCHA and requireCaptcha to disable/enable captcha in backend

Nathan-h05 committed:
updated code to follow Coding Conventions

JAlipes reviewed (changes requested):
In Mailer.ts make sure to use GetEnvOrFail to ensure we don't have missing env vars

<img width="674" height="291" alt="image" src="https://github.com/user-attachments/assets/c403f326-07ab-4703-a182-3fe1ffe78c3b" />

In Mailer.ts you can move the functions keep the functions but move the Emailoptions interface to the nodemailer.d.ts types or another file so we adhere to Seperations of concern.

<img width="824" height="823" alt="image" src="https://github.com/user-attachments/assets/d0718af8-1946-4e0a-809b-57afdf885fef" />

Same goes for VerifyTurnstile.ts you can create another types file or put it into nodemailer.d.ts

<img width="375" height="114" alt="image" src="https://github.com/user-attachments/assets/936d63da-6b27-4eaf-9f53-c5e2b83c8487" />

Everything else looks good just adhere to separation of concern

Nathan-h05 committed:
add user receipt email + address PR review fixes

- use GetEnvVarOrFail for envs (incl. CONTACT_REQUIRE_CAPTCHA); types moved to src/types/nodemailer.d.ts
- Adjusted rate limit to 5 requests per minute per IP
- Added confirmation email to submitter

Nathan-h05 committed:
Formatted spacing to 4 to match coding standards

Nathan-h05 committed:
Merge branch 'dev' into ContactFormSection

Nathan-h05 commented:
Addressed review feedback and resolved merge conflicts. Please re-review when you can.

JAlipes reviewed (approved):
Looks Good
